ls1intum/Artemis
# `Assessment`: Show statistics on course scores page

### Checklist
#### General


#### Client
### Motivation and Context


This PR introduces a score distribution similar to the exam statistics for the course score page. 
### Description

I refactored the exam score distribution into an own component in order to make it reusable for the course score. Besides that I added calculation logic for the mean of included in score and total points.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with some contributions

1. Log in to Artemis
2. Navigate to Course Administration
3. Click on the "Scores" button of the corresponding course
4. Inspect the new chart. make sure that the bar containing the included in score average is highlighted in light blue per default. Also make sure that this changes if you select the median via the checkbox.
5. Play around with grading keys and make sure the chart changes as expected in terms of intervals as well as in terms of coloring (similar to the exam score distribution)
     - If a non-bonus grading key exists, all bars representing a non-passing grade are colored red
     - If a bonus grading key exists, all bars representing scores that do not achieve a bonus point are colored yellow
     - If no grading key existis, all bars representing scores < 40% are colored yellow
     - Also check that the help icon tooltip next to the color legend makes sense
6. Make sure that the exam score distribution still works as expected

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





| File name | Branch coverage (%) | Line coverage (%) |
| -- | -- | -- |
| `course-scores.component.ts` | 73.72 | 95.06 |
| `exam-scores.component.ts` | 77.3 | 95.1 |
| `participant-scores-distribution.component.ts` |  85.96| 93.8 |
### Screenshots


![image](https://user-images.githubusercontent.com/80622272/154325166-a57aaca6-9eae-42ae-8671-b586854c0fd1.png)
